### PR TITLE
Add .NET 5 support information to .NET Tracer documentation

### PR DESCRIPTION
**Pull request by zacharycmontoya**
### What does this PR do?
This PR notes the following support in our .NET Tracer:
- Automatic instrumentation fully supports .NET 5 beginning with .NET Tracer version 1.20.0
- Custom instrumentation fully supports .NET 5 (no .NET Tracer version requirement)

### Motivation
.NET 5 is the upcoming release of .NET and it is being released on November 10. This docs update indicates our support of the platform and that the soon-to-be-released version 1.20.0 of the .NET Tracer has full automatic instrumentation support for the new platform.

### Preview
.NET Core Compatibility requirements: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-core/net5/tracing/setup_overview/compatibility_requirements/dotnet-core

.NET Core Automatic Instrumentation: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-core/net5/tracing/setup_overview/setup/dotnet-core/?tab=windows

.NET Framework Automatic Instrumentation: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-core/net5/tracing/setup_overview/setup/dotnet-framework?tab=environmentvariables

.NET Custom Instrumentation: https://docs-staging.datadoghq.com/zach.montoya/apm-dotnet-core/net5/tracing/setup_overview/custom_instrumentation/dotnet

### Additional Notes
The .NET versioning scheme is taking a new direction. There was legacy .NET Framework 1-4, and then .NET Core 1-3. From here on out, even though the tech is based on .NET Core 3, the platform will be named .NET 5, .NET 6, .NET 7, etc. Since it requires the same environment variables and setup as .NET Core, the setup information has just been added into the .NET Core pages instead of broken out into a separate page.

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/compatibility_requirements/dotnet-core.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ further_reading:
 
 The .NET Datadog Trace library is open source - view the [Github repository][1] for more information.
 
-The .NET Tracer supports automatic instrumentation on .NET Core 2.1 and 3.1. It also supports [.NET Framework][2].
+The .NET Tracer supports automatic instrumentation on .NET 5 (starting with .NET Tracer `1.20.0`) and on LTS .NET Core versions 2.1 and 3.1. It also supports [.NET Framework][2].
```
Comment (lucaspimentel) [inline content/en/tracing/compatibility_requirements/dotnet-core.md:13]: I don't think we generally mention specific tracer versions in the docs (it's implied that everything applies to the _current_ version), and LTS may be confusing for users ("is that a special non-standard version of .NET Core?"). Maybe we should keep it simple, like:

```suggestion
The .NET Tracer supports automatic instrumentation on .NET 5, .NET Core 3.1, and .NET Core 2.1. It also supports [.NET Framework][2].
```

File: content/en/tracing/custom_instrumentation/dotnet.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ This page details common use cases for adding and customizing observability with
 
 Add the `Datadog.Trace` [NuGet package][1] to your application. To create new spans, access the global tracer through the `Datadog.Trace.Tracer.Instance` property.
 
-Custom instrumentation is supported on **.NET Framework 4.5+** for Windows and on **.NET Core 2.1, 3.0, and 3.1** for Windows and Linux.
+Custom instrumentation is supported on **.NET Framework 4.5+** for Windows, on **.NET Core 2.0+** for Windows and Linux, and on **.NET 5** for Windows and Linux.
```
Comment (lucaspimentel) [inline content/en/tracing/custom_instrumentation/dotnet.md:29]: This adds .NET Core 2.0 and 2.2 as supported versions, which we didn't have listed before. Was this intentional?

File: content/en/tracing/setup/dotnet-core.md [hunk 3]
```diff
@@ -266,7 +266,7 @@ To manually instrument your code, add the `Datadog.Trace` [NuGet package][4] to
 
 For more details on manual instrumentation and custom tagging, see [Manual instrumentation documentation][5].
 
-Manual instrumentation is supported on .NET Framework 4.5 and above on Windows and on .NET Core 2.1, 3.0, and 3.1 on Windows and Linux.
+Manual instrumentation is supported on .NET Framework 4.5 and above on Windows, on .NET Core 2.0 and above on Windows and Linux, and on .NET 5 on Windows and Linux.
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:269]: Same comment about adding 2.0 and 2.2 as in `content/en/tracing/custom_instrumentation/dotnet.md`.

File: content/en/tracing/setup/dotnet-framework.md [hunk 4]
```diff
@@ -113,7 +113,7 @@ To manually instrument your code, add the `Datadog.Trace` [NuGet package][5] to
 
 For more details on manual instrumentation and custom tagging, see [Manual instrumentation documentation][6].
 
-Manual instrumentation is supported on .NET Framework 4.5 and above on Windows and on .NET Core 2.1, 3.0, and 3.1 on Windows and Linux.
+Manual instrumentation is supported on .NET Framework 4.5 and above on Windows, on .NET Core 2.0 and above on Windows and Linux, and on .NET 5 on Windows and Linux.
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-framework.md:116]: Same comment about adding 2.0 and 2.2 as in `content/en/tracing/custom_instrumentation/dotnet.md`.

File: content/en/tracing/setup/dotnet-core.md [hunk 5]
```diff
@@ -23,7 +23,7 @@ further_reading:
 
 ## Compatibility
 
-The .NET Tracer supports automatic instrumentation on .NET Core 2.1 and 3.1, as well as Microsoft-deprecated versions 2.0, 2.2 and 3.0.  For a full list of supported libraries, visit the [Compatibility Requirements][1] page.
+The .NET Tracer supports automatic instrumentation on .NET 5 (starting with .NET Tracer `1.20.0`), LTS .NET Core versions 2.1 and 3.1, and Microsoft-deprecated .NET Core versions 2.0, 2.2 and 3.0.  For a full list of supported libraries, visit the [Compatibility Requirements][1] page.
```
Comment (lucaspimentel) [inline content/en/tracing/setup/dotnet-core.md:26]: We should keep this in sync with the one in `content/en/tracing/compatibility_requirements/dotnet-core.md` (which doesn't mention "Microsoft-deprecated versions).

File: content/en/tracing/custom_instrumentation/dotnet.md (hunk 2 above)
Comment (zacharycmontoya) [reply content/en/tracing/custom_instrumentation/dotnet.md:29]: Yes this was intentional on my part. I figured that manual instrumentation (consuming the SDK) is completely supported by .NET Standard 2.0 so we shouldn't shy away from declaring support for it. The one thing that has conditional support is automatic instrumentation. Do you agree with this thought process?

Comment (lucaspimentel) [reply content/en/tracing/custom_instrumentation/dotnet.md:29]: Ok. My concern was that we don't actually run any tests on 2.0 or 2.2 If you feel confident about them, feel free to resolve my comment 😊

**Review by lucaspimentel — APPROVED** [review]

**Review by jaycdave88 — COMMENTED** [review]
lgtm

**Review by jaycdave88 — APPROVED** [review]

**Review by kayayarai — APPROVED** [review]

**Pull request merged**